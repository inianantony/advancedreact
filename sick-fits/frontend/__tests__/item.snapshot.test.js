import ItemComponent from '../components/item';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: "AABB",
    title: 'a cool title',
    price: 5000,
    description: "an amazing item",
    image: "small_img.jpg",
    largeImage: "large_img.jpg"
}

describe("<Item/>",()=>{
    it("renders and displays properly", ()=>{
        const wrapper = shallow(<ItemComponent item={fakeItem}/>);
        const priceTag = wrapper.find('PriceTag');
        const title = wrapper.find('Title a');
        const img = wrapper.find('img');

        expect(toJSON(wrapper)).toMatchSnapshot();
       
    });
});