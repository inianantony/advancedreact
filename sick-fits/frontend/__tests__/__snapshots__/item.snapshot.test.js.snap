// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and displays properly 1`] = `
<ItemStyles__Item>
  <img
    alt="a cool title"
    src="small_img.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "AABB",
          },
        }
      }
    >
      <a>
        a cool title
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    an amazing item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "path": "update",
          "query": Object {
            "id": "AABB",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="AABB"
    />
    <DeleteItem
      id="AABB"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
